src/main/java/org/springframework/data/couchbase/core/convert/translation/JsonValueConverter.java

```
@@ -0,0 +1,112 @@
+		try {
+			Class<? extends Enum> type = value.getClass();
+			return getJsonValueMethod(type).invoke(value);
+		} catch (IllegalAccessException | InvocationTargetException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	private Enum<?> getEnumConstant(Class<Enum> type, Object value) {
+		Map<Object, Enum<?>> enumConstants = enumConstantsCache.get(type);
+		if (enumConstants == null) {
+			enumConstants = new HashMap<>();
+			enumConstantsCache.put(type, enumConstants);
+			Method jsonValueMethod = getJsonValueMethod(type);
+			for (Enum<?> e : type.getEnumConstants()) { // cache all the constants while we're here
+				try {
+					enumConstants.put(jsonValueMethod.invoke(e), e);
+				} catch (IllegalAccessException | InvocationTargetException ex) {
+					throw new RuntimeException(ex);
+				}
+			}
+		}
+
+		Enum<?> theEnum = enumConstants.get(value);
+		if (theEnum == null) {
+			throw new RuntimeException("could not deserialize " + value + " to " + type);
+		}
+		return theEnum;
+	}
+
+	private Method getJsonValueMethod(Class<? extends Enum> type) {
+		Method jsonValueMethod = methodCache.get(type);
+		if (jsonValueMethod != null) {
+			return jsonValueMethod;
+		}
+		for (Method m : type.getDeclaredMethods()) {
+			JsonValue jsonValueAnn = m.getAnnotation(JsonValue.class);
+			if (jsonValueAnn != null && jsonValueAnn.value()) {
+				jsonValueMethod = m;
+				break;
+			}
+		}
+		if (jsonValueMethod == null) { // Is this necessary? Useful?
+			for (Method m : type.getMethods()) {
+				JsonValue jsonValueAnn = m.getAnnotation(JsonValue.class);
+				if (jsonValueAnn != null && jsonValueAnn.value()) {
+					jsonValueMethod = m;
+					break;
+				}
+			}
+		}
+		// if nothing was found, use toString()
+		if (jsonValueMethod == null) {
+			try {
+				jsonValueMethod = type.getMethod("toString");
+			} catch (NoSuchMethodException e) {
+				throw new RuntimeException(e);
+			}
```

If there was no enabled @ JsonValue method, instead use toString().  This will be ok for serialization, but deserialization will fail.  Maybe we should just throw an exception, as it is impossible to get here without the converter factory finding an @JsonValue method that is enabled.